taskcluster/docker/node/build.py

```diff
@@ -65,6 +81,30 @@ def get_hash(path, hash_alg="sha256"):
     return h.hexdigest()
 
 
+def check_manifest(path):
+    xpi = ZipFile(path, "r")
+    manifest = {}
+    _found = False
+    for manifest_name in ("manifest.json", "webextension/manifest.json"):
+        try:
+            with xpi.open(manifest_name, "r") as f:
+                manifest = json.load(f)
+        except KeyError:
+            print(f"{manifest_name} doesn't exist in {path}...")
+            continue
+        for _key in ("applications", "browser_specific_settings"):
+            _id = manifest.get(_key, {}).get("gecko", {}).get("id", None)
+            if _id is None:
+                continue
```

Comment:
Can this happen with add-ons handled by `shipit` currently ?

Reply:
I think we'll break in the signing step. This block would catch it earlier, in the build step.